### ISSUE TITLE
IcebergDataSink fails when writing empty batch with append transaction

### ISSUE THREAD

**Issue by osipovartem** [JanKaul/iceberg-rust#171]
When attempting to write data through the IcebergDataSink, if the input logical plan produces an empty result (e.g., due to a filter that excludes all rows), the sink successfully writes empty Parquet files. However, during the commit phase, when trying to create an Iceberg append transaction, the operation fails with the following error:

```rust
Empty data and delete files doesn't have the right format
```

**Steps to Reproduce:**
1.	Execute a query with a filter that results in an empty dataset:
```sql
CREATE TABLE target_table AS 
SELECT * 
FROM source_table 
WHERE 1 = 0;
```
2.	The logical plan executes without rows, but still reaches the write phase.
3.	IcebergDataSink writes empty output files.
4.	During the append commit, an exception is raised.

**Expected Behavior:**
If the result is empty, no files should be written and the append transaction should either:
	•	**be skipped entirely (i.e., treated as a no-op), or**
	•	be safely committed without attempting to include empty or invalid file metadata.

**Actual Behavior:**
The append transaction fails due to an invalid state with empty data files.

**Suggested Fix:**
Before committing, IcebergDataSink should check if any valid data files were written. If none exist, it should either:
	•	skip the append transaction, or
	•	handle the empty state in a way that’s compatible with Iceberg table format rules.

**Comment by JanKaul** [comment]
Thanks a lot for raising this issue. And this only occurred after upgrading to DF 47?

I think its best to not even write the parquet files.

I'll look into it.

**Comment by osipovartem** [comment]
yes, after upgrading to DF 47

**Comment by osipovartem** [comment]
Related issue https://github.com/apache/datafusion/issues/16001

**Comment by JanKaul** [comment]
Did you check if the parquet files are actually written to disk? Because I don't know if the [AsyncArrowWriter](https://docs.rs/parquet/latest/parquet/arrow/async_writer/struct.AsyncArrowWriter.html) writes bytes to disk even before any recordbatch is flushed. So even in case there are no record batches I'm not sure if it starts writing some parquet header or something.

Otherwise, I think it's reasonable to just return with an empty update vector instead of throwing the error.

**Comment by osipovartem** [comment]
Yes, the parquet is empty
![Image](https://github.com/user-attachments/assets/8cf149bc-3a59-426a-8068-0f880952cde2)
![Image](https://github.com/user-attachments/assets/7c5fa0e5-ff24-4b9e-8095-6723315fe11d)

Parquet raw data

<img width="971" alt="Image" src="https://github.com/user-attachments/assets/295cc9d1-58ab-443e-9cf6-56500c2cfb04" />
